lib/src/windows/file_picker_windows.dart

```diff
@@ -94,78 +95,94 @@ class FilePickerWindows extends FilePicker {
-        FILEOPENDIALOGOPTIONS.FOS_FORCEFILESYSTEM |
-        FILEOPENDIALOGOPTIONS.FOS_NOCHANGEDIR;
-    hr = fileDialog.setOptions(options);
-    if (!SUCCEEDED(hr)) throw WindowsException(hr);
+      final options = optionsPointer.value |
+          FILEOPENDIALOGOPTIONS.FOS_PICKFOLDERS |
+          FILEOPENDIALOGOPTIONS.FOS_FORCEFILESYSTEM |
+          FILEOPENDIALOGOPTIONS.FOS_NOCHANGEDIR;
+      hr = fileDialog.setOptions(options);
+      if (!SUCCEEDED(hr)) throw WindowsException(hr);
+    } finally {
+      free(optionsPointer);
+    }
 
-    final title = TEXT(dialogTitle ?? defaultDialogTitle);
-    hr = fileDialog.setTitle(title);
-    if (!SUCCEEDED(hr)) throw WindowsException(hr);
-    free(title);
+    final title = TEXT(dialogTitle ?? 'Select a Folder');
+    try {
+      hr = fileDialog.setTitle(title);
+      if (!SUCCEEDED(hr)) throw WindowsException(hr);
+    } finally {
+      free(title);
+    }
 
     if (initialDirectory != null) {
       final folder = TEXT(initialDirectory);
       final riid = convertToIID(IID_IShellItem);
       final ppv = calloc<Pointer>();
-      hr = SHCreateItemFromParsingName(folder, nullptr, riid, ppv);
-      final item = IShellItem(ppv.cast());
-      free(riid);
-      free(folder);
-      if (!SUCCEEDED(hr)) throw WindowsException(hr);
-      hr = fileDialog.setFolder(item.ptr.cast<Pointer<COMObject>>().value);
-      if (!SUCCEEDED(hr)) throw WindowsException(hr);
-    }
 
-    final hwndOwner = lockParentWindow ? GetForegroundWindow() : NULL;
-    hr = fileDialog.show(hwndOwner);
-    if (!SUCCEEDED(hr)) {
-      CoUninitialize();
+      try {
+        hr = SHCreateItemFromParsingName(folder, nullptr, riid, ppv);
+        if (!SUCCEEDED(hr) || ppv.value == nullptr) {
+          throw WindowsException(hr);
+        }
 
-      if (hr == HRESULT_FROM_WIN32(WIN32_ERROR.ERROR_CANCELLED)) {
-        return Future.value(null);
+        final item = IShellItem(ppv.cast());
+        hr = fileDialog.setFolder(item.ptr.cast<Pointer<COMObject>>().value);
+        if (!SUCCEEDED(hr)) throw WindowsException(hr);
+      } finally {
+        free(riid);
+        free(folder);
+        free(ppv);
       }
-      throw WindowsException(hr);
     }
 
-    final ppsi = calloc<COMObject>();
-    hr = fileDialog.getResult(ppsi.cast());
-    if (!SUCCEEDED(hr)) throw WindowsException(hr);
+    hr = fileDialog.show(NULL);
+    if (!SUCCEEDED(hr)) return null;
 
-    final item = IShellItem(ppsi);
-    final pathPtr = calloc<Pointer<Utf16>>();
-    hr = item.getDisplayName(SIGDN.SIGDN_FILESYSPATH, pathPtr);
-    if (!SUCCEEDED(hr)) throw WindowsException(hr);
-
-    final path = pathPtr.value.toDartString();
+    final ppv = calloc<Pointer<COMObject>>();
+    String? selectedPath;
+    try {
+      hr = fileDialog.getResult(ppv);
+      if (!SUCCEEDED(hr)) throw WindowsException(hr);
 
-    hr = item.release();
-    if (!SUCCEEDED(hr)) throw WindowsException(hr);
+      final item = IShellItem(ppv.cast());
```

Comment:
Do we need to release the shell item?

Reply:
That's a good question and honestly I'm not expert enough in this kind of code to give an accurated answer. I guess it’s not necessary as it's a wrapper around ppv which is the allocated pointer.